### PR TITLE
permissions updated

### PR DESCRIPTION
**Pull request by VikashHart**
[Link to Ticket 🎟](_paste_link_here_)

## Description
[//]: # (Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)
permissions updated

## Type of change

[//]: # (Please delete options that are not relevant.)

- [ ] 🐞Bug fix (non-breaking change which fixes an issue)
- [ ] 🌟New feature (non-breaking change which adds functionality)
- [ ] 🚧Breaking change (fix or feature that would cause existing functionality to not work as expected)

## UI Changes 
[//]: # (Add images/gifs if applicable)

Before            | After
----------------  | -----------------
image/gif before  |  image / gif after

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Manually tested
- [ ] Unit tests added/updated

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Update my ticket with relevant information (inProgress or any comments needed)

### CODE REVIEW

File: PhotoBooth/Camera Page/CameraViewController.swift [hunk 1]
```diff
@@ -84,11 +84,32 @@ class CameraViewController: UIViewController {
...
 
+    private func requestCameraAccess() {
+        let alertController = UIAlertController(title: "Looks like camera access is denied",
+                                                message: "In order to take pictures Lens needs access to your camera. To update your app permissions, click settings below.",
+                                                preferredStyle: .alert)
+        alertController.addAction(UIAlertAction(title: "Cancel", style: .destructive))
+        alertController.addAction(UIAlertAction(title: "Settings", style: .cancel) { _ in
+            if let url = URL(string: UIApplicationOpenSettingsURLString) {
+                UIApplication.shared.open(url, options: [:], completionHandler: { _ in
+                    // Handle
```
Comment (vhart) [inline PhotoBooth/Camera Page/CameraViewController.swift]: what is this?

File: PhotoBooth/Camera Page/CameraViewController.swift [hunk 2]
```diff
@@ -84,11 +84,32 @@ class CameraViewController: UIViewController {
...
+        present(alertController, animated: true)
+    }
+
     private func presentConfigurationCard() {
         let setupCardPartialModal = SetUpCardPartialModal(onConfigureFinalized: { [weak self] configuration, modal in
-            modal.dismiss()
-            self?.configureShoot(config: configuration)
-            self?.presentSwipeToCancelPrompt()
+            guard let strongSelf = self else { return }
+            if strongSelf.viewModel.checkCameraAccess() {
```
Comment (vhart) [inline PhotoBooth/Camera Page/CameraViewController.swift]: checkCameraAccess does not imply a return of bool. The name of a function should describe its functionality.

File: PhotoBooth/Camera Page/CameraViewController.swift [hunk 3]
```diff
@@ -84,11 +84,32 @@ class CameraViewController: UIViewController {
...
+                modal.dismiss()
+                self?.configureShoot(config: configuration)
+                self?.presentSwipeToCancelPrompt()
+            } else {
+                self?.requestCameraAccess()
```
Comment (vhart) [inline PhotoBooth/Camera Page/CameraViewController.swift]: this is a misnomer. We don't request camera access. `presentMissingCameraAccessAlert`. also you unwrapped self as strong self but you keep referring to self.

File: PhotoBooth/Camera Page/CameraViewControllerViewModel.swift [hunk 4]
```diff
@@ -80,6 +82,29 @@ class CameraViewControllerViewModel: CameraViewControllerViewModeling {
         }
     }
 
+    func checkCameraAccess() -> Bool {
+        switch AVCaptureDevice.authorizationStatus(for: .video) {
+        case .denied:
+            print("Denied, request permission from settings")
```
Comment (vhart) [inline PhotoBooth/Camera Page/CameraViewControllerViewModel.swift]: why are these print statements still here?

File: PhotoBooth/Camera Page/CameraViewController.swift (hunk 1 above)
Comment (VikashHart) [reply PhotoBooth/Camera Page/CameraViewController.swift]: that's how we open the settings, I can get rid of the completion handler

File: PhotoBooth/Camera Page/CameraViewControllerViewModel.swift (hunk 4 above)
Comment (vhart) [reply PhotoBooth/Camera Page/CameraViewControllerViewModel.swift]: you can combine these cases into 
case .denied, .restricted, .notDetermined: false
case .authorized: return true

you need to do a self review

**Pull request merged**